ext/js/data/sandbox/anki-note-data-creator.js

```diff
@@ -485,15 +519,17 @@ export class AnkiNoteDataCreator {
 
     /**
      * @param {import('dictionary').TermDictionaryEntry} dictionaryEntry
-     * @returns {import('anki-templates').TermPronunciation[]}
+     * @returns {import('anki-templates').TermPitchAccent[]}
      */
     _getTermPitches(dictionaryEntry) {
         // eslint-disable-next-line @typescript-eslint/no-this-alias
         const self = this;
         const results = [];
         const {headwords} = dictionaryEntry;
-        for (const {headwordIndex, dictionary, dictionaryIndex, dictionaryPriority, pitches} of dictionaryEntry.pronunciations) {
+        for (const {headwordIndex, dictionary, dictionaryIndex, dictionaryPriority, pronunciations} of dictionaryEntry.pronunciations) {
             const {term, reading} = headwords[headwordIndex];
+            /** @type {import('dictionary').PitchAccent[]} */
+            const pitches = pronunciations.filter((v) => v.type === 'pitch-accent');
```

Comment:
To fix this, just do like this
```js
            /** @type {import('dictionary').PitchAccent[]} */
            const pitches = [];
            for (const pronunciation of pronunciations) {
                if (pronunciation.type !== 'pitch-accent') { continue; }
                pitches.push(pronunciation);
            }
```

Reply:
Same for other locations. Create a utility function for doing this. This can probably be templatized, I'll look into how to easily do that.

Reply:
```js
// ext/js/data/sandbox/anki-note-data-creator.js

// No type annotation necessary before declaration
const pitches = this._getPronunciationsOfType(pronunciations, 'pitch-accent');


// ext/js/data/sandbox/anki-note-data-creator.js

/**
 * @template {import('dictionary').PronunciationType} T
 * @param {import('dictionary').Pronunciation[]} pronunciations
 * @param {T} type
 * @returns {import('dictionary').PronunciationGeneric<T>[]}
 */
_getPronunciationsOfType(pronunciations, type) {
    /** @type {import('dictionary').PronunciationGeneric<T>[]} */
    const results = [];
    for (const pronunciation of pronunciations) {
        if (pronunciation.type === type) { continue; }
        // This is type safe, but for some reason the cast is needed.
        results.push(/** @type {import('dictionary').PronunciationGeneric<T>} */ (pronunciation));
    }
    return results;
}


// types/ext/dictionary.d.ts

export type PronunciationType = Pronunciation['type'];

export type PronunciationGeneric<T extends PronunciationType> = Extract<Pronunciation, {type: T}>;
```